cogeo_mosaic/overviews.py

```diff
@@ -124,80 +128,67 @@ def create_low_level_cogs(
-                                    f"No asset for tile {x}-{y}-{base_zoom}"
-                                )
-
-                            if assets:
-                                tile, mask = mosaic_tiler(
-                                    assets,
-                                    x,
-                                    y,
-                                    base_zoom,
-                                    cogeo.tile,
-                                    indexes=info[6],
-                                    tilesize=tilesize,
-                                    pixel_selection=defaults.FirstMethod(),
-                                )
-
-                            return window, tile, mask
-
-                        with futures.ThreadPoolExecutor(
-                            max_workers=threads
-                        ) as executor:
-                            future_work = [
-                                executor.submit(_get_tile, item) for item in blocks
-                            ]
-                            with click.progressbar(
-                                futures.as_completed(future_work),
-                                length=len(future_work),
-                                show_percent=True,
-                            ) as future:
-                                for res in future:
-                                    pass
-
-                        for f in _filter_futures(future_work):
-                            window, tile, mask = f
+                # with rasterio_open(in_memory, out_path, **params) as mem:
+                with rasterio.open(out_path, "w", **params) as dst:
+
+                    def _get_tile(wind):
+                        idx, window = wind
+                        x = extrema["x"]["min"] + idx[1]
+                        y = extrema["y"]["min"] + idx[0]
+                        t = mercantile.Tile(x, y, base_zoom)
+
+                        kds = set(find_quadkeys(t, mosaic.quadkey_zoom))
+                        if not mosaic_quadkeys.intersection(kds):
+                            return window, None, None
+
+                        assets = mosaic.tile(*t)
+                        if not assets:
+                            return window, None, None
+
+                        tile, mask = mosaic_tiler(
+                            assets,
+                            x,
+                            y,
+                            base_zoom,
+                            cogeo.tile,
+                            indexes=info["indexes"],
+                            tilesize=tilesize,
+                            pixel_selection=defaults.FirstMethod(),
+                        )
+                        return window, tile, mask
+
+                    with futures.ThreadPoolExecutor(max_workers=threads) as executor:
+                        future_work = executor.map(_get_tile, blocks)
+                        for result in tqdm(future_work, total=len(blocks)):
```

Comment:
👋 any reason to use tqdm instead of click progressbar ?

Reply:
I couldn't get it to show with `click.progressbar`. Maybe it was something with `futures` that I just didn't understand, but I had to switch to `tqdm` to get some sort of progress bar to show